src/PackagePrivate/ParserValidator.php

```diff
@@ -20,58 +23,55 @@ public function __construct(Parser $parser)
 
     public function isValid(): bool
     {
-        $methods = get_class_methods(__CLASS__);
-        $messages = [];
+        $this->validateInput();
+        $this->validateSeason();
 
-        foreach($methods as $method){
-            if(false !== strpos($method, 'validate') && 'validate' !== $method){
-                try{
-                    call_user_func([$this, $method]);
-                }catch (\Exception $e){
-                    $messages[] = $e->getMessage();
-                }
-            }
-        }
-
-        $this->messages = $messages;
-        return 0 === count($messages);
+        return 0 === count($this->messages);
     }
 
-    public function validateInput(): void
+    private function validateInput(): void
     {
         $parser = $this->parser;
         $input = $parser->getInput();
         $matches = $parser->getMatches();
 
         $hasValue = false;
+        /** @var mixed $v
+         * Need this to satisfy Psalm check
+         */
         foreach($matches as $k => $v){
-            assert(is_string($v));
+            if(!is_string($v)) {
+                $this->messages[] = "Invalid data format: $k must be a string";
+                break;
+            }
             if("" != $v){
                 $hasValue = true;
             }
         }
 
         if(!$hasValue){
-            throw new \InvalidArgumentException(sprintf(
-                'Invalid edtf format "%s".',$input
-            ));
+            $this->messages[] = "Invalid edtf format $input";
         }
     }
 
-    public function validateSeason(): void
+    private function validateSeason(): void
     {
         $season = $this->parser->getSeason();
-        if($season > 0){
-            if(false === ($season >= 21 && $season <= 41)){
-                throw new \InvalidArgumentException(sprintf(
-                    'Invalid season number "%s" in "%s" is out of range. Accepted season number is between 21-41',
-                    $season,
-                    $this->parser->getInput()
-                ));
-            }
+
+        if($season > 0 && $this->isOutsideValidRange($season)){
+            $this->messages[] = "Invalid season number $season in {$this->parser->getInput()} is out of range. Accepted season number is between 21-41";
         }
     }
 
+    /**
+     * @param int $season
+     * @return bool
+     */
```

Comment:
Nice method extraction!

I think it good practice to not add comments like these if all the info is already expressed in PHP itself